docs/spec/plugins.md

```diff
@@ -0,0 +1,117 @@
+Plugins
+=======
+
+Specification
+-------------
+
+NSM provides a way to extend its functionality by creating plugins. Here is a small glossary:
```

Comment:
NSM is cluster agnostic technology and any cluster specific functionality could be provided via extensibility plugins mechanism.